pkg/resolver/search.go

```diff
@@ -157,38 +162,44 @@ func (s *SearchResult) buildSearchQuery(ctx context.Context, count bool, uid boo
 		ds = goqu.From(schemaTable, jsb)
 	}
 
-	//SELECT CLAUSE
-	if count {
-		selectDs = ds.Select(goqu.COUNT("uid"))
-	} else if uid {
-		selectDs = ds.Select("uid")
-	} else {
-		selectDs = ds.SelectDistinct("uid", "cluster", "data")
-	}
-
-	//WHERE CLAUSE
 	if s.input != nil && (len(s.input.Filters) > 0 || (s.input.Keywords != nil && len(s.input.Keywords) > 0)) {
-		whereDs = WhereClauseFilter(s.input)
-		sql, _, err := selectDs.Where(whereDs...).ToSQL()
-		klog.V(3).Info("Search query before adding RBAC clause:", sql, " error:", err)
-		//RBAC CLAUSE
-		if s.userData != nil {
-			_, userInfo := rbac.GetCache().GetUserUID(ctx)
-			whereDs = append(whereDs,
-				buildRbacWhereClause(ctx, s.userData, userInfo)) // add rbac
-		} else {
-			panic(fmt.Sprintf("RBAC clause is required! None found for search query %+v for user %s ", s.input,
-				ctx.Value(rbac.ContextAuthTokenKey)))
+		//WHERE CLAUSE
+		whereDs, s.propTypes, err = WhereClauseFilter(s.context, s.input, s.propTypes)
+		if whereDs != nil {
+
+			//SELECT CLAUSE
+			if count {
+				selectDs = ds.Select(goqu.COUNT("uid"))
+			} else if uid {
+				selectDs = ds.Select("uid")
+			} else {
+				selectDs = ds.SelectDistinct("uid", "cluster", "data")
+			}
+
+			sql, _, err = selectDs.Where(whereDs...).ToSQL() //use original query
+			klog.V(3).Info("Search query before adding RBAC clause:", sql, " error:", err)
+
+			//RBAC CLAUSE
+			if s.userData != nil {
+				_, userInfo := rbac.GetCache().GetUserUID(ctx)
+				whereDs = append(whereDs,
+					buildRbacWhereClause(ctx, s.userData, userInfo)) // add rbac
+			} else {
+				panic(fmt.Sprintf("RBAC clause is required! None found for search query %+v for user %s ", s.input,
+					ctx.Value(rbac.ContextAuthTokenKey)))
+			}
+		} else if err != nil {
+			klog.Errorf("Error building Search query: %s", err.Error())
 		}
+	} else {
+		klog.Errorf("Error building Search query: %s", err.Error())
```

Comment:
Please update this log.
```suggestion
		klog.Errorf("Error: query input must contain a filter or keyword. Received: %+v", s.input)
```

Reply:
Resolved with this commit. https://github.com/stolostron/search-v2-api/pull/91/commits/1666e7a9398314d715ba987f746e1f245ceffcf9